infra/modules/providers/azure/service-plan/main.tf

```diff
@@ -0,0 +1,27 @@
+resource "random_id" "remotestate_account_name" {
```

Comment:
We should avoid business logic for randomizing resource names in terraform modules. This should be up to module clients. JFYI, [here's ](https://github.com/Microsoft/bedrock/blob/master/test/bedrock_Azure_common_kv_test.go#L18)how bedrock randomizes resource names for clusters.